### PR TITLE
183189703 sort events services assessments

### PR DESCRIPTION
**Pull request by ZachCMP**
https://www.pivotaltracker.com/story/show/183189703

I implemented this as concerns as I've done for other object types (clients, organizations, etc.)

### CODE REVIEW

**Review by eanders — APPROVED** [review]

File: drivers/hmis/app/graphql/types/hmis_schema/has_events.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      end
+
+      def resolve_events(scope = object.events, **args)
+        apply_event_arguments(scope, **args)
+      end
+
+      private
+
+      def apply_event_arguments(scope, user: current_user, sort_order: :date)
+        events_scope = scope.viewable_by(user)
```
Comment (gigxz) [inline drivers/hmis/app/graphql/types/hmis_schema/has_events.rb]: Hmm, I'm getting errors when I try to query for events/services/assessments now. Here is the query I tried and the error:

```
{
  enrollment(id: "46762") {
    id
    events(sortOrder: DATE) {
      nodes {
        id
      }
    }
    services(sortOrder: DATE) {
      nodes {
        id
      }
    }
    assessments(sortOrder: DATE) {
      nodes {
        id
      }
    }
  }
}

```


```
{
  "errors": [
    {
      "message": "undefined method `viewable_by' for :events:Symbol",
      "backtrace": [
        "/app/drivers/hmis/app/graphql/types/hmis_schema/has_events.rb:36:in `apply_event_arguments'",
        "/app/drivers/hmis/app/graphql/types/hmis_schema/has_events.rb:30:in `resolve_events'",
        "/app/drivers/hmis/app/graphql/types/hmis_schema/enrollment.rb:51:in `events'",
```

File: drivers/hmis/app/models/hmis/hud/assessment.rb [hunk 2]
```diff
@@ -15,4 +15,17 @@ class Hmis::Hud::Assessment < Hmis::Hud::Base
   use_enum :assessment_types_enum_map, ::HUD.assessment_types
   use_enum :assessment_levels_enum_map, ::HUD.assessment_levels
   use_enum :prioritization_statuses_enum_map, ::HUD.prioritization_statuses
+
+  SORT_OPTIONS = [:date].freeze
```
Comment (gigxz) [inline drivers/hmis/app/models/hmis/hud/assessment.rb]: Can we call it `assessment_date`? we may want to sort by other dates later on, like last updated date etc.

File: drivers/hmis/app/models/hmis/hud/event.rb [hunk 3]
```diff
@@ -16,4 +16,17 @@ class Hmis::Hud::Event < Hmis::Hud::Base
   use_enum :referral_result_enum_map, ::HUD.referral_results
   use_common_enum :prob_sol_div_rr_result_enum_map, :yes_no_missing
   use_common_enum :referral_case_manage_after_enum_map, :yes_no_missing
+
+  SORT_OPTIONS = [:date].freeze
```
Comment (gigxz) [inline drivers/hmis/app/models/hmis/hud/event.rb]: Same here, lets call it `event_date`

File: drivers/hmis/app/models/hmis/hud/service.rb [hunk 4]
```diff
@@ -32,4 +32,17 @@ class Hmis::Hud::Service < Hmis::Hud::Base
   use_enum :s_s_v_f_sub_type3_enum_map, ::HUD.s_s_v_f_sub_type3_map
   use_enum :s_s_v_f_sub_type4_enum_map, ::HUD.s_s_v_f_sub_type4_map
   use_enum :s_s_v_f_sub_type5_enum_map, ::HUD.s_s_v_f_sub_type5_map
+
+  SORT_OPTIONS = [:date].freeze
```
Comment (gigxz) [inline drivers/hmis/app/models/hmis/hud/service.rb]:
```suggestion
  SORT_OPTIONS = [:date_provided].freeze
```

**Comment by ZachCMP** [comment]
@gigxz Fixed all the items you mentioned

File: drivers/hmis/app/graphql/types/hmis_schema/has_events.rb [hunk 5]
```diff
@@ -32,9 +32,8 @@ def resolve_events(scope = object.events, **args)
 
       private
 
-      def apply_event_arguments(scope, user: current_user, sort_order: :date)
-        events_scope = scope.viewable_by(user)
-        events_scope.sort_by_option(sort_order) if sort_order.present?
+      def apply_event_arguments(scope, sort_order: :event_date)
+        scope.sort_by_option(sort_order) if sort_order.present?
```
Comment (gigxz) [inline drivers/hmis/app/graphql/types/hmis_schema/has_events.rb:36]: I'm still seeing the same error, now it's just `undefined method sort_by_option for :events:Symbol`. Are you able to test this in graphiql @ZachCMP?

Comment (ZachCMP) [reply drivers/hmis/app/graphql/types/hmis_schema/has_events.rb:36]: I had trouble doing that because I have no events in my local DB

**Comment by ZachCMP** [comment]
@gigxz I think I found the problem. I was calling the `resolve_*` method with the wrong args. Also neither `Event`, `Service`, nor `Assessment` have a `viewable_by` scope, so I removed that part from all the concerns.

**Review by gigxz — APPROVED** [review]

**Pull request merged**